app/controllers/jobseekers/job_applications/qualifications_controller.rb

```diff
@@ -49,28 +50,52 @@ def form_attributes
     when "select_category"
       {}
     when "edit"
-      qualification.slice(:category, :finished_studying, :finished_studying_details, :grade, :institution, :name, :subject, :year)
+      attributes = qualifications.first.slice(:category, :finished_studying, :finished_studying_details, :institution, :name, :year)
+      (qualifications + built_qualifications).map.with_index { |qualification, index|
+        attributes.merge!(qualification.slice(:subject, :grade).transform_keys { |key| key + (index + 1).to_s })
+      }.uniq
+      attributes.merge(qualification_params.to_h)
     when "create", "update", "submit_category"
       qualification_params
     end
   end
 
+  def built_qualifications
```

Comment:
I'm trying quite hard to understand this code, but it doesn't feel very readable and seems like quite a complex method

Reply:
I'll split it in two and give those methods understandable names.

Reply:
(Refactored)